### ISSUE TITLE
Security improvements based on Security Slam 2023 requirements

### ISSUE THREAD

**Issue by eddie-knight** [metal3-io/baremetal-operator#1399]
Hello metal3-io community!

In case you missed it, last week was the kickoff for the month-long Cloud Native Security Slam! You can read more about the event and find the kickoff webinar recording [here](https://community.cncf.io/cloud-native-security-slam/).

In the kickoff last week, it was announced that there are 5 event badges that projects can pursue. There will be swag prizes and a $100 gift card awarded for the first 200 badges that are competed— that's up to $500 per project!

To help with work prioritization, we completed a survey of CNCF end users across multiple industries, including Construction, Cybersecurity, Aerospace & Defense, Game Development & Consumer Services, Consulting, and Nuclear. 

Through this survey, end users have identified their interest in seeing security improvements to the projects they use. We've asked them to share which Security Slam goals are most interesting to them— and we've compiled the results in a hope that this will help your prioritization during the upcoming event.

While some users have not authorized us to share their name, we've still included their responses in our calculation for you. We CAN tell you that the metal3-io responses included **Epic Games**.

After calculating the responses according to the interest-weight, we've found these to be the most interesting things that metal3-io end users would like to see, from the five possible Security Slam badges.

1. The Mechanizer
2. The Defender
3. The Cleaner

You can read more about the success metrics and find resources to help you achieve them [here](https://github.com/security-slam/metrics/tree/main/cncf/2023).

Join us in the **#security-slam** channel on the [CNCF Slack](https://slack.cncf.io/) to ask questions and get help from the community!

**Comment by Rozzii** [comment]
Hey thanks for thinking about us!

I am not sure we have the manpower at the moment to participate on the event, this info has come on a short notice but in relation to "The Mechanizer"  we are working on it but this is a long process for us so I am not sure we would fit into the time scale of the event. Related to the two other that is built around the CLOMonitor, we might take a look at the future but as far as I know that is not part of our security roadmap @tuminoid .

In any case the action points (achievements) of this event will serve as good checklist for our project!!!

Thanks once again!  @eddie-knight

**Comment by eddie-knight** [comment]
Thanks for the update @Rozzii! 

FYI, some projects have found success by creating help wanted issues for tasks that could be actioned by community members. If you'd like, you can generate markdown checklists for your project using the https://clomonitor.io interface.

**Comment by zaneb** [comment]
baremetal-operator [using the `code-lite` checkset](https://github.com/cncf/clomonitor/blob/main/data/cncf.yaml#L1618) instead of `code` in [CLOMonitor](https://clomonitor.io/projects/cncf/metal3-io) looks like something we could/should fix.

**Comment by tuminoid** [comment]
Organization-wise, we can and should act on each of the badge checklists. LIke @rozzii said we're working on the general posture and have improved quite a lot in many aspects in the past year, but especially on SBOM side we have a lot to do. 

We have documented the starting situation ~year ago here with the OpenSSF Scorecard analysis: https://github.com/Nordix/metal3-clusterapi-docs/blob/main/security/openssf-scorecards.md and have worked to fix those issues. It is soon time for the yearly update and I think we could add similar document for the CLOmonitor to work off.

**Comment by Rozzii** [comment]
As we have not participated I have renamed the ticket bu I am inclined to accept this as a general issue to track the related security improvements.

**Comment by tuminoid** [comment]
> baremetal-operator [using the `code-lite` checkset](https://github.com/cncf/clomonitor/blob/main/data/cncf.yaml#L1618) instead of `code` in [CLOMonitor](https://clomonitor.io/projects/cncf/metal3-io) looks like something we could/should fix.

I'm updating the Metal3 CLOMonitor config here: https://github.com/cncf/clomonitor/pull/1377

TL;DR: 
- BMO: changed code-lite -> code
- IPAM: added "code"
- docs: changed community -> docs
- community: added "community"
- ironic-image: added "code-lite"

**Comment by tuminoid** [comment]
...And the update is merged. Let's see next week how we fare with the new checks in place.

**Comment by tuminoid** [comment]
> ...And the update is merged. Let's see next week how we fare with the new checks in place.

We are solid here, current score is 84.

/close
